### DISCUSSION TITLE
Problema com rotas em projeto AngularJS + Wordpress

### DISCUSSION THREAD

**Discussion by yurischulz** [frontendbr/forum#895] [Perguntas e Respostas]
Fala galera, tudo certo?

Estou desenvolvendo um projeto utilizando WordPress (4.9.2) e AngularJS (1.6.8) e estou com dificuldade de configurar as rotas utilizando o html5Mode. 

Quando eu rodo a aplicação, eu consigo navegar pelas rotas e tudo funciona perfeitamente, porém quando eu atualizo a página me retorna um erro 404 pelo WordPress. Fiz diversas Googladas e testei várias soluções, mas nenhuma resolveu este problema. Em teoria, a solução seria adicionar uma regra usando a função `add_rewrite_rule`.

Eis que vem o problema, segue abaixo parte do meu código:

### AngularJS
As minhas rotas no AngularJS estão configuradas da seguinte forma:

`$routeProvider.when('/inicio/', {`
`templateUrl: path + '/assets/src/js/partials/homepage.html',`
`controller: 'homeController'`
`});`
`$locationProvider.html5Mode({ enabled: true, requireBase: true, rewriteLinks: 'internal-link' }).hashPrefix('');`

### WordPress
Permalinks:
`/index.php/%postname%/`

Rewrite_rule:
`add_action( 'init', 'wpd_foo_rewrite_rule' );`
`function wpd_foo_rewrite_rule() {`
`	add_rewrite_rule('^(.?.+?)?(:/([0-9]+))?/?$', 'index.php/$matches[1]/', 'top');`
`}`

Eu utilizei um plugin chamado monkeyman-rewrite-analyzer, que basicamente me da uma lista de expressões regulares baseadas em cada rota do WordPress, para montar a função acima. Mas nada de resolver este problema.

Alguém consegue me dar uma luz?

Obrigado :)

**Comment by felipefialho** [comment]
@woliveiras Você que tem skills em ng1, sabe ajudar aqui?
